lib/tictactoe/board.ex

```diff
@@ -1,3 +1,21 @@
 defmodule TicTacToe.Board do
-  defstruct cells: Enum.to_list(1..9)
+  alias TicTacToe.Board
+
+  defstruct cells: nil
+
+  def update(index, marker, %Board{cells: cells}) do
+    cells
+    |> List.replace_at(index, marker)
+    |> then(&%Board{cells: &1})
+  end
+
+  def space_occupied?(index, %Board{cells: cells}, markers) do
+    cells
+    |> Enum.at(index)
+    |> then(&is_marker?(markers, &1))
```

Comment:
Non-blocking, but since `is_marker?` is something we own, we could just swap the order of the arguments so that we don't need to use `then` here